clippy_lints/src/methods/mod.rs

```diff
@@ -1911,6 +1959,42 @@ fn lint_filter_map<'a, 'tcx>(
     }
 }
 
+/// lint use of `filter_map().next()` for `Iterators`
+fn lint_filter_map_next<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, expr: &'tcx hir::Expr, filter_args: &'tcx [hir::Expr]) {
+    if match_trait_method(cx, expr, &paths::ITERATOR) {
+        let msg = "called `filter_map(p).next()` on an `Iterator`. This is more succinctly expressed by calling \
+                   `.find_map(p)` instead.";
+        let filter_snippet = snippet(cx, filter_args[1].span, "..");
+        if filter_snippet.lines().count() <= 1 {
```

Comment:
Why only print the note, when the snippet is a one liner? Did it cause problems with the snippet being multiple lines?

Reply:
I thought there was a reason behind that, because other parts of this file are doing the same treatment. I took it out to see how it would look like and updated the .stderr files.